src/ContainerDetails.jsx

```diff
@@ -6,38 +6,38 @@ const moment = require('moment');
 const _ = cockpit.gettext;
 
 const render_container_state = (container) => {
-    if (container.status === "running") {
-        return cockpit.format(_("Up since $0"), moment(container.createdat, utils.GOLANG_TIME_FORMAT).calendar());
+    if (container.State === "running") {
+        return cockpit.format(_("Up since $0"), moment(container.StartedAt * 1000).calendar());
     }
     return cockpit.format(_("Exited"));
 };
 
-const render_container_published_ports = ({ ports }) => {
+const render_container_published_ports = (ports) => {
```

Comment:
That looks weird -- does React have some magic to flatten the property object if there's just one property?

Reply:
Not sure what you mean. I just pass into this function `container.Ports` which is array. (I changed how it used to be called as well, I find this cleaner approach)

Reply:
OK, now *I* don't understand what you mean.. I see nothing getting passed on here, `ports` is just iterated over in the `for` loop and that didn't change at all. Normally such a functional-style component is being used  either as `props => { props.foo }` or more robustly as `({ foo, bar, ... })  => { bar }`. That's why this syntax is utterly confusing -- `ports` would now be an object that *contains* properties like "ports", but it is not the property itself. ... unless React has some magic if a component just gets a single property. (But if it does, let's please not use it ... I find this utterly confusing)

Reply:
`<dd>{render_container_published_ports(container.Ports)}</dd>`
This function is called with array (possibly empty one).

Reply:
Oh! Sorry, this isn't actually a React component, more like a helper function. I see it now, sorry for the noise.